presto-main/src/main/java/com/facebook/presto/dynamicCatalog/DynamicCatalogController.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    @Path("add")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response addCatalog(CatalogVo catalogVo)
+    {
+        try {
+            log.info("addCatalog : input values are " + catalogVo);
+            ConnectorId connectorId = connectorManager.createConnection(catalogVo.getCatalogName(), catalogVo.getConnectorName(), catalogVo.getProperties());
+            updateConnectorIdAnnouncement(announcer, connectorId, internalNodeManager);
+            log.info("addCatalog() : catalogConfigurationDir: " + catalogConfigurationDir.getAbsolutePath());
+            writeToFile(catalogVo);
+            log.info("addCatalog() : Successfully added catalog " + catalogVo.getCatalogName());
+            return successResponse(responseParser.build("Successfully added catalog: " + catalogVo.getCatalogName(), 200));
+        }
+        catch (Exception ex) {
+            log.error("addCatalog() : Error adding catalog " + ex.getMessage());
+            return failedResponse(responseParser.build("Error adding Catalog: " + ex.getMessage(), 500));
+        }
+    }
+
+    @Path("delete")
+    @DELETE
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteCatalog(@QueryParam("catalogName") String catalogName)
+    {
+        log.info("deleteCatalog(): Deleting catalog: " + catalogName);
+        String responseMessage = "";
+        if (catalogManager.getCatalog(catalogName).isPresent()) {
+            log.info("deleteCatalog() : Catalog exists so deleting catalog " + catalogName);
+            connectorManager.dropConnection(catalogName);
+            responseMessage = (deletePropertyFile(catalogName) == true) ? "Successfully deleted" : "Error deleting catalog";
+        }
+        else {
+            log.info("deleteCatalog() : Catalog doesn't exists, Can't be deleted " + catalogName);
+            return failedResponse(responseParser.build("Catalog doesn't exists: " + catalogName, 500));
+        }
+        log.info("deleteCatalog() : successfully deleted catalog " + catalogName);
+        return successResponse(responseParser.build(responseMessage + " : " + catalogName, 200));
+    }
+
+    private boolean deletePropertyFile(String catalogName)
+    {
+        return new File(getPropertyFilePath(catalogName)).delete();
+    }
+
+    private void writeToFile(CatalogVo catalogVo)
+            throws Exception
+    {
+        final Properties properties = new Properties();
+        properties.put("connector.name", catalogVo.getConnectorName());
+        properties.putAll(catalogVo.getProperties());
+        String filePath = getPropertyFilePath(catalogVo.getCatalogName());
+        log.info("filepath: " + filePath);
+        File propertiesFile = new File(filePath);
+        try (OutputStream out = new FileOutputStream(propertiesFile)) {
```

Comment:
Small nit: you might consider using Files.newOuputStream here instead. See https://github.com/prestodb/presto/pull/12614 for details about why File{Input,Output}Stream should be avoided in cases where any {Input,Output}Stream will do.